### PR TITLE
Add FlPointerManager to process pointer events from GTK in a form suitable for Flutter.

### PR DESCRIPTION
**Pull request by robert-ancell**
This matches FlScrollingManager and FlKeyboardManager.

Add tests for this behaviour that was previously missing.

### CODE REVIEW

**Comment by robert-ancell** [comment]
@mattkae @hbatagelo trying to add you as reviewers too, but it's not allowing me. Perhaps you need some more memberships from https://github.com/flutter/flutter/blob/master/docs/contributing/Contributor-access.md?

**Comment by hbatagelo** [comment]
> @mattkae @hbatagelo trying to add you as reviewers too, but it's not allowing me. Perhaps you need some more memberships from https://github.com/flutter/flutter/blob/master/docs/contributing/Contributor-access.md?

I think we'll first need the "contributor access" status as mentioned in the document. Could you be our sponsor and ask on `#server-support` to make us contributors? I'm not sure if this alone is sufficient, but it appears to be a prerequisite.

File: shell/platform/linux/fl_pointer_manager.cc [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+#include "flutter/shell/platform/linux/fl_pointer_manager.h"
+
+#include "flutter/shell/platform/linux/fl_engine_private.h"
+
+static constexpr int kMicrosecondsPerMillisecond = 1000;
+
+struct _FlPointerManager {
+  GObject parent_instance;
+
+  // Engine events are being sent to.
```
Comment (cbracken) [inline shell/platform/linux/fl_pointer_manager.cc:14]: It looks like there might be more to this comment than is currently here.

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

lgtm with a couple nits.

**Pull request merged**